dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -1216,6 +1224,19 @@
 
 			<!-- Link in the output from ILC -->
 			<_NativeExecutableObjectFiles Include="$(NativeObject)" />
+
+			<!-- By default perform aggressive attribute trimming, except when explicitly disabled -->
+			<_TrimmerFeatureSettings Condition="'$(_OriginalAggressiveAttributeTrimming)' != 'false' and '%(Identity)' == 'System.AggressiveAttributeTrimming'">
+				<Value>true</Value>
+				<Trim>true</Trim>
+			</_TrimmerFeatureSettings>
+			<RuntimeHostConfigurationOption Condition="'$(_OriginalAggressiveAttributeTrimming)' != 'false' and '%(Identity)' == 'System.AggressiveAttributeTrimming'">
+				<Value>true</Value>
+				<Trim>true</Trim>
+			</RuntimeHostConfigurationOption>
+
+			<!-- Explicitly root the framework assembly due to NativeAOT trimming incompatibility tracked: https://github.com/dotnet/runtime/issues/86649 -->
+			<TrimmerRootAssembly Include="Microsoft.iOS" />
```

Comment:
```suggestion
			<TrimmerRootAssembly Include="$(_PlatformAssemblyName)" />
```

Reply:
The name is different for each platform, so this needs to be adjusted. Aside from that, is there a longer description why is this necessary? I suppose with #18519 and #18529 this would no longer be necessary, correct?

Reply:
Thank you for the correction, it was an oversight.

> I suppose with https://github.com/xamarin/xamarin-macios/pull/18519 and https://github.com/xamarin/xamarin-macios/pull/18529 this would no longer be necessary, correct?

If I am not mistaken, there is still an issue with ILC not knowing about `[Preserve]`. The issue I noted in the comment should track/document all these incompatibilities.

Reply:
> If I am not mistaken, there is still an issue with ILC not knowing about `[Preserve]`.

I am almost sure that `[Preserve]` in the platform assembly is used only for the constructors that would get rooted by the generated code in #18519.